src/Storages/StorageDistributed.cpp

```diff
@@ -946,7 +1057,10 @@ QueryTreeNodePtr buildQueryTreeDistributed(SelectQueryInfo & query_info,
     const StorageSnapshotPtr & distributed_storage_snapshot,
     const StorageID & remote_storage_id,
     const ASTPtr & remote_table_function,
-    const ASTPtr & additional_filter = nullptr)
+    const ASTPtr & additional_filter = nullptr,
+    const NameSet * columns_to_cast = nullptr,
```

Comment:
Why a pointer instead of `const NameSet & columns_to_cast = {}`? I think the same can be done for the others - if not, consider using `std::optional`